disco_aws_automation/disco_deploy.py

```diff
@@ -318,10 +319,11 @@ def run_tests_with_maintenance_mode(self, ami):
         This method puts all AMIs not matching the passed in AMI into maintenance mode.
         If tests pass the old instances are left in maintenance mode, otherwise they are returned to normal.
         '''
-        self._set_maintenance_mode(self._get_old_instances(ami.id), True)
+        hostclass = self._disco_bake.ami_hostclass(ami)
```

Comment:
I'll change this to access ```ami_hostclass``` statically like we do in other places in the code base.